src/services/transaction_watcher_signer_service.ts

```diff
@@ -0,0 +1,364 @@
+                logger.warn({
+                    message:
+                        'unsticking of transaction skipped as the targetGasPrice is less than or equal to the gas price it was submitted with',
+                    txHash: tx.txHash,
+                    txGasPrice: tx.gasPrice,
+                    targetGasPrice,
+                });
+                continue;
+            }
+            const signer = this._getSignerByPublicAddressOrThrow(tx.from);
+            try {
+                await this._unstickTransactionAsync(tx, targetGasPrice, signer);
+            } catch (err) {
+                logger.error(`failed to unstick transaction ${tx.txHash}`, { err });
+            }
+        }
+    }
+    private async _checkForConfirmedTransactionsAsync(): Promise<void> {
+        // we are checking for transactions that are already in the included
+        // state, but can potentially be affected by a blockchain reorg.
+        const latestBlockNumber = await this._web3Wrapper.getBlockNumberAsync();
+        const transactionsToCheck = await this._transactionRepository.find({
+            where: { status: TransactionStates.Included },
+        });
+        if (transactionsToCheck.length === 0) {
+            return;
+        }
+        for (const tx of transactionsToCheck) {
+            if (tx.txHash === undefined || tx.blockNumber === undefined) {
+                logger.error({
+                    mesage: 'transaction that has an included status is missing a txHash or blockNumber',
+                    refHash: tx.refHash,
+                    from: tx.from,
+                });
+                continue;
+            }
+            const txInBlockchain = await this._web3Wrapper.getTransactionByHashAsync(tx.txHash);
+            if (txInBlockchain === undefined) {
+                // transaction that was previously included is not identified by
+                // the node, we change its status to submitted and see whether
+                // or not it will appear again.
+                tx.status = TransactionStates.Submitted;
+                await this._transactionRepository.save(tx);
+                continue;
+            }
+            if (txInBlockchain.blockNumber === null) {
+                // transaction that was previously included in a block is now
+                // showing without a blockNumber, but exists in the mempool of
+                // an ethereum node.
+                tx.status = TransactionStates.Mempool;
+                tx.blockNumber = undefined;
+                await this._transactionRepository.save(tx);
+                continue;
+            } else {
+                if (tx.blockNumber !== txInBlockchain.blockNumber) {
+                    logger.warn({
+                        message:
+                            'transaction that was included has a different blockNumber stored than the one returned from RPC',
+                        previousBlockNumber: tx.blockNumber,
```

Comment:
I think one of these is coming back as a `BigNumber` and the other is a `number`

Reply:
Yes, handled similarly to the nonce.